### ISSUE TITLE
Intermittent travis test failure

### ISSUE THREAD

**Issue by yadudoc** [Parsl/parsl#145]
The following is a test failure snippet from Travis. This test battery runs successfully on my laptop, and also passed on the py3.6 tests, and has only failed on py3.5.

If the test_3 in test_docs_futures failed with delta 13.87, it's most likely from the machine being slow or from previous apps delaying the execution of time critical apps.

======================================================================
FAIL: parsl.tests.test_threads.test_docs_futures.test_3
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.5.5/lib/python3.5/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/travis/build/Parsl/parsl/parsl/tests/test_threads/test_docs_futures.py", line 70, in test_3
    assert delta < 5, "Took too much time"
AssertionError: Took too much time
-------------------- >> begin captured stdout << ---------------------
20
delta :  13.874757289886475
--------------------- >> end captured stdout << ----------------------
-------------------- >> begin captured logging << --------------------
parsl.dataflow.dflow: INFO: Task 2 submitted for App wait_sleep_double, waiting on tasks []
parsl.dataflow.dflow: INFO: Task 2 launched on site 0
parsl.dataflow.dflow: INFO: Task 3 submitted for App wait_sleep_double, waiting on tasks []
parsl.dataflow.dflow: INFO: Task 3 launched on site 0
parsl.dataflow.dflow: INFO: Task 4 submitted for App wait_sleep_double, waiting on tasks [2, 3]
parsl.dataflow.dflow: INFO: Task 2 completed with <Future at 0x7efe3c62d5f8 state=finished returned int>
parsl.dataflow.dflow: INFO: Task 3 completed with <Future at 0x7efe3c62d710 state=finished returned int>
parsl.dataflow.dflow: INFO: Task 4 launched on site 0
parsl.dataflow.dflow: INFO: Task 4 completed with <Future at 0x7efe3c62da20 state=finished returned int>
parsl.dataflow.dflow: INFO: Task 0 completed with <Future at 0x7efe3c624e80 state=finished returned int>
--------------------- >> end captured logging << ---------------------

**Comment by annawoodard** [comment]
@yadudoc this is the same as #143

**Comment by annawoodard** [comment]
Sure, I'll mark it as superseded by this one.

**Comment by annawoodard** [comment]
@yadudoc fec5f7b looks like pep-8 changes, which commit are you referring to?

**Comment by yadudoc** [comment]
For reference, this is what I suspect was happening:
``parsl/parsl/tests/test_threads/test_docs_futures`` contains 5 tests or so. The tests use apps that sleep, and ``test_1( )`` launches an app without waiting for it. ``test_3( )`` that follows launches a few tasks and uses total time to completion to indirectly guarantee that multiple threads were active. The app from test_1 occassionally interfered with test_3 resulting in failures. This is a hunch. Solution is to have test_1 wait for the app.

**Comment by yadudoc** [comment]
@annawoodard Sorry pasted in the wrong commit, I've edited my earlier comment.

**Comment by annawoodard** [comment]
Sounds good. I am worried that no matter what, relying on time measurements cannot be fully deterministic/independent of machine performance/independent of machine load. But I'm happy to defer worrying about that unless it causes more problems.

**Comment by annawoodard** [comment]
Another problem with relying on time measurements is that not only false negatives but also false positives in the case there is a big performance difference between the original and subsequent testing machines.

**Comment by yadudoc** [comment]
We are running sleep tasks to avoid machine performance problems. Generally if you sleep more than ~100ms, timing is reliable.

**Comment by annawoodard** [comment]
Here is another example which should be fixed as part of this:
[https://api.travis-ci.org/v3/job/356533671/log.txt](https://api.travis-ci.org/v3/job/356533671/log.txt). It looks like this is a similar issue in `tests/test_threads/test_docs_futures.py`.
